pkg/xpkg/find.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xpkg
+
+import (
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+const (
+	errNoMatch    = "directory does not contain a compiled crossplane package"
+	errMultiMatch = "directory contains multiple compiled crossplane packages"
+)
+
+// FindXpkgInDir finds compiled Crossplane packages in a directory.
+func FindXpkgInDir(fs afero.Fs, root string) (string, error) {
+	f, err := fs.Open(root)
+	if err != nil {
+		return "", err
+	}
+	defer func() { _ = f.Close() }()
+	files, err := f.Readdir(-1)
+	if err != nil {
+		return "", err
+	}
+	path := ""
+	for _, file := range files {
+		match, err := filepath.Match(XpkgMatchPattern, file.Name())
+		if err != nil {
+			return "", err
+		}
```

Comment:
```suggestion
		// Match only returns an error if XpkgMatchPattern is malformed.
		match, _ := filepath.Match(XpkgMatchPattern, file.Name())  //nolint:errcheck
```

> The only possible returned error is ErrBadPattern, when pattern is malformed.

Usually I'd recommend using the `MustMatch` variant to panic here since the only way we'd hit an error is if the program was fundamentally broken, but none exists in this case. I suggest we simply don't handle this error.